internal/services/notification/runwebhookdeliverys.go

```diff
@@ -0,0 +1,215 @@
+// Copyright 2023 Sorint.lab
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package notification
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/sorintlab/errors"
+
+	"agola.io/agola/internal/sqlg/lock"
+	"agola.io/agola/internal/sqlg/sql"
+	"agola.io/agola/services/notification/types"
+)
+
+const (
+	MaxRunWebhookDeliverysLimit = 40
+	RunWebhookDeliverysLockKey  = "runwebhookdeliveryevents"
+)
+
+func (n *NotificationService) RunWebhookDeliverysHandlerLoop(ctx context.Context) {
+	for {
+		if err := n.runWebhookDeliverysHandler(ctx); err != nil {
+			n.log.Err(err).Send()
+		}
+
+		sleepCh := time.NewTimer(1 * time.Second).C
+		select {
+		case <-ctx.Done():
+			return
+		case <-sleepCh:
+		}
+	}
+}
+
+func (n *NotificationService) runWebhookDeliverysHandler(ctx context.Context) error {
+	var curSequence uint64
+
+	err := n.d.Do(ctx, func(tx *sql.Tx) error {
+		// start from last event
+		runWebhookDelivery, err := n.d.GetLastRunWebhookDelivery(tx)
```

Comment:
Why are you starting from the last sequence? What happens if previous deliveries haven't been handled?